crates/autopilot/src/run_loop.rs

```diff
@@ -784,17 +771,27 @@ impl RunLoop {
                 })?;
                 onchain_task.await
             }
-        }
+        };
+
+        // Clean up the in-flight orders regardless the result.
+        self.in_flight_orders
+            .lock()
+            .await
+            .retain(|order| !solved_order_uids.contains(order));
```

Comment:
Moving it here doesn't guarantee the execution anymore. The following code might return before that.
https://github.com/cowprotocol/services/pull/3003/files#diff-8a11a8276d7572d1f949773ad29d0e2089fc86498483ca91e6f6828cac310517R768-R771

Reply:
Good catch. Fixed. I still think there is no need for `wait_for_transaction` to be a separate function.